### PR TITLE
Firedrake applications

### PR DESCRIPTION
**Pull request by sv2518**
I added research papers, which cite the Firedrake paper "Firedrake: Automating the Finite Element Method by Composing Abstractions" and use it for their application. I only looked at publications from 2019 until today.

### CODE REVIEW

File: docs/source/_static/firedrake-apps.bib [hunk 1]
```diff
@@ -738,6 +991,26 @@ @Article{Pan2019
...
+               Galbraith, Marshall C and Caplan, Philip and Carson, Hugh A and
+               Michal, Todd R and Krakos, Joshua A and Kamenetskiy, Dmitry S and
+               Loseille, Adrien and others},
+  booktitle = {AIAA Scitech 2019 Forum},
+  pages =     {1723},
+  year =      {2019}
+} 
+
+@PhdThesis{Pembery2019,
+  title =   {The Helmholtz Equation in Heterogeneous and Random Media:
```
Comment (wence-) [inline docs/source/_static/firedrake-apps.bib:1006]: Capitalisation.

File: docs/source/_static/firedrake-apps.bib [hunk 2]
```diff
@@ -738,6 +991,26 @@ @Article{Pan2019
...
+             Analysis and Numerics},
+  author =  {Pembery, OWEN RHYS},
```
Comment (wence-) [inline docs/source/_static/firedrake-apps.bib:1008]: Remove excessive caps here!

File: docs/source/_static/firedrake-apps.bib [hunk 3]
```diff
@@ -881,6 +1198,27 @@ @Misc{Wimmer2019
...
+
+@Article{Xia2020,
+  title =     {Nonlinear bifurcation analysis of stiffener profiles
+               via deflation techniques},
+  author =    {Xia, Jingmin and Farrell, Patrick E and Castro, Saullo GP},
+  journal =   {Thin-Walled Structures},
+  volume =    {149},
+  pages =     {106662},
+  year =      {2020},
+  publisher = {Elsevier}
```
Comment (wence-) [inline docs/source/_static/firedrake-apps.bib:1219]: `doi = {10.1016/j.tws.2020.106662}`

File: docs/source/_static/firedrake-apps.bib [hunk 4]
```diff
@@ -365,6 +510,19 @@ @Book{Gibson2019
   url =          {https://doi.org/10.1007/978-3-030-23957-2}
 }
 
+@Article{Gibson2020,
```
[the remark this answers is not in the document]
Comment (sv2518) [reply docs/source/_static/firedrake-apps.bib:513]: The doi for that changed, is that right?

[the remark this answers is not in the document]
Comment (wence-) [reply docs/source/_static/firedrake-apps.bib]: I think so because it moved from GMDD to GMD.

File: docs/source/_static/bibliography.bib [hunk 5]
```diff
@@ -18,29 +18,31 @@ @Article{Bercea2016
...
-  author =       {Thomas H. Gibson and Lawrence Mitchell and David A.
-                  Ham and Colin J. Cotter},
-  title =        {{A domain-specific language for the hybridization
-                  and static condensation of finite element methods}},
-  year =         2018,
-  archiveprefix ={arXiv},
-  eprint =       {1802.00303},
-  primaryclass = {cs.MS},
-  url =          {https://arxiv.org/abs/1802.00303}
+@Article{Gibson2020,
```
Comment (wence-) [inline docs/source/_static/bibliography.bib:21]: Sorry, can we also keep the preprint links where available?

**Comment by wence-** [comment]
> FYI most of the new additions here don't have DOIs

Most of them are preprints!

**Pull request merged**